src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_rpc_plugin.cc

```diff
@@ -32,41 +32,57 @@
 
 #include "sdl_rpc_plugin/sdl_rpc_plugin.h"
 #include "sdl_rpc_plugin/sdl_command_factory.h"
+#include "sdl_rpc_plugin/extensions/system_capability_app_extension.h"
 
 namespace sdl_rpc_plugin {
+namespace app_mngr = application_manager;
 namespace plugins = application_manager::plugin_manager;
 
-bool SDLRPCPlugin::Init(
-    application_manager::ApplicationManager& app_manager,
-    application_manager::rpc_service::RPCService& rpc_service,
-    application_manager::HMICapabilities& hmi_capabilities,
-    policy::PolicyHandlerInterface& policy_handler) {
+CREATE_LOGGERPTR_GLOBAL(logger_, "SdlRPCPlugin")
+
+bool SDLRPCPlugin::Init(app_mngr::ApplicationManager& app_manager,
+                        app_mngr::rpc_service::RPCService& rpc_service,
+                        app_mngr::HMICapabilities& hmi_capabilities,
+                        policy::PolicyHandlerInterface& policy_handler) {
   command_factory_.reset(new sdl_rpc_plugin::SDLCommandFactory(
       app_manager, rpc_service, hmi_capabilities, policy_handler));
   return true;
 }
 
 bool SDLRPCPlugin::IsAbleToProcess(
     const int32_t function_id,
-    const application_manager::commands::Command::CommandSource
-        message_source) {
+    const app_mngr::commands::Command::CommandSource message_source) {
   return command_factory_->IsAbleToProcess(function_id, message_source);
 }
 
 std::string SDLRPCPlugin::PluginName() {
   return "SDL RPC Plugin";
 }
 
-application_manager::CommandFactory& SDLRPCPlugin::GetCommandFactory() {
+app_mngr::CommandFactory& SDLRPCPlugin::GetCommandFactory() {
   return *command_factory_;
 }
 
-void SDLRPCPlugin::OnPolicyEvent(
-    application_manager::plugin_manager::PolicyEvent event) {}
+void SDLRPCPlugin::OnPolicyEvent(plugins::PolicyEvent event) {}
 
 void SDLRPCPlugin::OnApplicationEvent(
-    application_manager::plugin_manager::ApplicationEvent event,
-    application_manager::ApplicationSharedPtr application) {}
+    plugins::ApplicationEvent event,
+    app_mngr::ApplicationSharedPtr application) {
+  if (plugins::ApplicationEvent::kApplicationRegistered == event) {
+    application->AddExtension(
+        std::make_shared<SystemCapabilityAppExtension>(*this, *application));
+  } else if (plugins::ApplicationEvent::kDeleteApplicationData == event) {
+    DeleteSubscriptions(application);
+  }
+}
+
+void SDLRPCPlugin::DeleteSubscriptions(app_mngr::ApplicationSharedPtr app) {
+  auto& ext = SystemCapabilityAppExtension::ExtractExtension(*app);
+  auto subscriptions = ext.Subscriptions();
+  for (auto& sub : subscriptions) {
+    ext.unsubscribeFrom(sub);
```

Comment:
Is it necessary to loop through the subscriptions when there is a function that does unsubscribeFromAll()?